### ISSUE TITLE
Nav Unification: Smaller Viewport: How to Switch Sites?

### ISSUE THREAD

**Issue by ktyfuller604** [Automattic/wp-calypso#48182]
#### Steps to reproduce

1. Reduce screen size to 600px or less
2. Click on Reader, then Click on My SItes (brings you to the default site)

#### What I expected

An indicator that I could click on My Sites to switch sites, or a Switch Sites menu option. 

#### What happened instead

No indication of the fact I could switch sites. Although clicking on "My Sites" did bring up the site switcher, on the full desktop view there is an arrow indicating an option. 

#### Browser / OS version

MacOS Catalina v 10.15.7/ Chrome v. Version 86.0.4240.198

#### Screenshot / Video

https://user-images.githubusercontent.com/1398304/103698869-de486400-4f56-11eb-917e-e5644ad6702d.mov

#### Context / Source

#manual-testing

**Comment by cpapazoglou** [comment]
The site switcher belongs to the sidebar of **My Sites** section. Not knowing how the sidebar opens is a side effect of removing that navigational element
![](https://cln.sh/7xonSs+)

As of now, **My Sites**, **Reader**, **Me** act both for switching to the according section AND for opening the sidebar.
![SS 2021-01-05 at 17 03 17](https://user-images.githubusercontent.com/12430020/103661940-faa8c880-4f77-11eb-8dc8-bf99cf381c9c.gif)

I agree that is somehow counter intuitive. @sfougnier any thoughts here?

**Comment by ghost** [comment]
Thanks for the ping @cpapazoglou 

I agree that the current behavior on mobile feels disorienting. My expectation would be that if I selected a nav item in the masterbar, it would toggle open the corresponding menu rather than immediately take me to a page with the menu hidden. Adjusting the flow to display the menu first may help alleviate the confusion around switching sites as well as general confusion around how to access navigation throughout the experience.

https://user-images.githubusercontent.com/67021691/103800672-af93c180-501a-11eb-8195-3eadb8fdb04a.mov

Within tablet, upon selecting 'Reader', the sidebar would be displayed until a user has made a selection within it.
<img width="768" alt="iPad mini - 1" src="https://user-images.githubusercontent.com/67021691/103801561-e1f1ee80-501b-11eb-8b6c-8d65d125dc58.png">

**Comment by cpapazoglou** [comment]
@sfougnier actually this a great idea! Let me rephrase it to make sure I understood it correctly.

**At any circumstance, "My Sites", "Reader", "Me" will act us "open / close" for their respective sidebar.**

I see a UX problem here though, while in `wp-admin` we cannot open a sidebar for "Reader" or "Me", we can only redirect them to the calypso page.

From development perspective, it will not be easy. Right now each section has each own unique sidebar and ignores the other two. If we are OK with the UX problem, then we will research how we can develop it.

**Comment by ghost** [comment]
> "My Sites", "Reader", "Me" will act us "open / close" for their respective sidebar.

Yes.

> I see a UX problem here though, while in wp-admin we cannot open a sidebar for "Reader" or "Me", we can only redirect them to the calypso page.

Ok, I see. So to clarify, there would be no way to redirect to the calypso page with the sidebar toggled open?

If the answer to that is still no, maybe we keep the menu icon bar that already exists in Calypso (what you have in your screenshots above). It deviates from core, but this is a specific use case and it's clearer than assigning the masterbar icons 2 types of interactions. Would we be able to maintain the visibility of this bar within wp-admin screens?

<img width="1977" alt="Screen Shot 2021-01-07 at 10 52 55 AM" src="https://user-images.githubusercontent.com/67021691/103913447-8b95b600-50d6-11eb-9fb3-9dc94ac00a42.png">

[View in Figma](https://www.figma.com/file/9cy0DIowQkPAu0rodb9jsL/WP-Admin-for-Dotcom?node-id=1305%3A73978)

**Comment by cpapazoglou** [comment]
> Ok, I see. So to clarify, there would be no way to redirect to the calypso page with the sidebar toggled open?

We can make that happen, sure.

Let me recap:

**Desktop** (both calypso & wp-admin):
- Clicking "My Sites", "Reader", "Me" switches to the respective screen

**Mobile** (wp-admin):
- Clicking "My Sites" opens the sidebar. 
- Clicking "Reader", "Me" redirects to calypso with the respective sidebar open

**Mobile** (calypso):
- Clicking "My Sites", "Reader", "Me" opens the respective sidebar. We may need to change this to: switches to the respective screen AND opens the sidebar, would that be ok? This is both easier from development perspective & more consistent with the Desktop behavior and mobile (wp-admin) behavior

**Comment by ghost** [comment]
Yes, this sounds good! It would be great to test it out to make sure it feels right.

> We may need to change this to: switches to the respective screen AND opens the sidebar, would that be ok? This is both easier from development perspective & more consistent with the Desktop behavior and mobile (wp-admin) behavior

Also yes, and I like the idea of it being more consistent.
